Makefile

```diff
@@ -1,10 +1,18 @@
-BIN_OUTPUT_PATH = bin/$(shell uname -s)-$(shell uname -m)
+SOURCE_OS ?= $(shell uname -s | tr '[:upper:]' '[:lower:]')
+SOURCE_ARCH ?= $(shell uname -m)
+TARGET_OS ?= $(SOURCE_OS)
+TARGET_ARCH ?= $(SOURCE_ARCH)
+ifeq ($(TARGET_ARCH),aarch64)
+    TARGET_ARCH = arm64
+else ifeq ($(TARGET_ARCH),x86_64)
+    TARGET_ARCH = amd64
+endif
```

Comment:
Do we also need to do this aliasing for `SOURCE_ARCH`?

Reply:
We do not need it but can add it for consistency.